app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -46,6 +47,10 @@ def ems_inv_to_hashes
 
       private
 
+      def get_network_interfaces
+        @data[:network_interfaces] = gather_data_for_this_region(@nis)
```

Comment:
@djberg96 
The profile for every NIC on every instance in the subscription is stored in the @data hash which is carried through post processing. This will not cause the worker to bomb out because the post processing logic will simply ignore the ;network_interfaces hash key. However, Im wondering how expensive storing all this data will be in large scale environments. Can you create a new instance hash variable in the Azure refresh_parser since you do not need the network_interface profiles beyond the parser?
Other than this one suggestion I think this refactoring is very nice improvement. :+1: